pkgs/applications/version-management/mercurial/default.nix

```diff
@@ -20,13 +20,13 @@ in python3Packages.buildPythonApplication rec {
   patches = [
     # https://phab.mercurial-scm.org/D10638, needed for below patch to apply
     (fetchpatch {
-      url = "https://phab.mercurial-scm.org/file/data/oymk4awh2dd7q6cwjbzu/PHID-FILE-bfcr7qrp5spg42wspxpd/D10638.diff";
-      sha256 = "0mfi324is02l7cnd3j0gbmg5rpyyqn3afg3f73flnfwmz5njqa5f";
+      url = "https://phab.mercurial-scm.org/file/data/l7p2r4zcctcr3pzlybv2/PHID-FILE-bwjzxlz6sbegk3s4irik/D10638.diff";
+      sha256 = "07bhprh5zwl37snn59jsvx0mnp6s0n3hi0zl8zwbb20nlbvfbvfz";
     })
     # https://phab.mercurial-scm.org/D10639, fixes https://bz.mercurial-scm.org/show_bug.cgi?id=6514
     (fetchpatch {
-      url = "https://phab.mercurial-scm.org/file/data/re4uqdhtknjiacx2ogwu/PHID-FILE-4m26id65dno5gzix2ngh/D10639.diff";
-      sha256 = "0h5ilrd2x1789fr6sf4k1mcvxdh0xdyr94yawdacw87v3x12c8cb";
+      url = "https://phab.mercurial-scm.org/file/data/v53nhburhtkhpccyecei/PHID-FILE-6v34oll6r2gkqo4ng5nt/D10639.diff";
+      sha256 = "1azj90r4xc4ygdli0bv1d4yi0nl5dx0vbiri3ihlfzaykhqgl3w9";
```

Comment:
```suggestion
      sha256 = "0h5ilrd2x1789fr6sf4k1mcvxdh0xdyr94yawdacw87v3x12c8cb";
```

Reply:
```
nix-prefetch-url "https://phab.mercurial-scm.org/file/data/v53nhburhtkhpccyecei/PHID-FILE-6v34oll6r2gkqo4ng5nt/D10639.diff"
[0.0 MiB DL]
path is '/nix/store/yka0g62n515qh2k6y8lli8cx3vpclfiw-D10639.diff'
1azj90r4xc4ygdli0bv1d4yi0nl5dx0vbiri3ihlfzaykhqgl3w9
``` weird

Reply:
`fetchpatch` is different from `fetchurl`. `fetchpatch` normalizes the patch by cleaning it up (removing comments and other useless info).